db/test/fns/worker_manager_test.js

```diff
@@ -43,12 +43,26 @@ suite(testing.suiteName(), function() {
       wp.owner || 'me@me.com',
       wp.email_on_error || false,
     ];
-    const with_cap = await db.fns.update_worker_pool_with_capacity(...input);
+    const with_cap = await db.fns.update_worker_pool_with_capacity_and_counts_by_state(...input);
     for (const wp of with_cap) {
-      assert(wp.current_capacity !== undefined);
-      delete wp.current_capacity;
+      assert(wp.requested_count !== undefined);
+      assert(wp.running_count !== undefined);
+      assert(wp.stopping_count !== undefined);
+      assert(wp.stopped_count !== undefined);
+      assert(wp.requested_capacity !== undefined);
+      assert(wp.running_capacity !== undefined);
+      assert(wp.stopping_capacity !== undefined);
+      assert(wp.stopped_capacity !== undefined);
+      delete wp.requested_count;
+      delete wp.running_count;
+      delete wp.stopping_count;
+      delete wp.stopped_count;
+      delete wp.requested_capacity;
+      delete wp.running_capacity;
+      delete wp.stopping_capacity;
+      delete wp.stopped_capacity;
```

Comment:
I see the original test only checked if the capacity was `!== undefined` rather than something more restrictive - is there value in the test being that the value is a positive integer, for example? I'm not entirely sure how the test framework works, and whether that adds any value or not (perhaps the db schema definition is sufficient to ensure that it is an integer, if it isn't defined), but on initial glance the check looks like it might be a little permissive.